### ISSUE TITLE
`Commands.addMissingOptionalToParam` can't be set in type evaluator

### ISSUE THREAD

**Issue by heejaechang** [microsoft/pyright#6571]
it looks like this code path

https://github.com/microsoft/pyright/blob/5148c9de9d3ff8e7102b2281e3baf8e5cb60476a/packages/pyright-internal/src/analyzer/typeEvaluator.ts#L17305-L17311

will never reach since if

https://github.com/microsoft/pyright/blob/main/packages/pyright-internal/src/analyzer/typeEvaluator.ts#L17294

pass then `isNoneWithoutOptional` will be always `false`

for it to be `true`

https://github.com/microsoft/pyright/blob/main/packages/pyright-internal/src/analyzer/typeEvaluator.ts#L17258

this code path should be taken

https://github.com/microsoft/pyright/blob/main/packages/pyright-internal/src/analyzer/typeEvaluator.ts#L17563

but then, 

https://github.com/microsoft/pyright/blob/main/packages/pyright-internal/src/analyzer/typeEvaluator.ts#L17294

if condition will always return `false`

code example.

```python
def foo(a: int = None):
     pass
```

by the way, not sure why the code action is only enabled when `strictParameterNoneValue` is `false`. it feels like it should be opposite since that's when it shows diagnostic.

**Comment by heejaechang** [comment]
it looks like code snippet thing doesn't work with type eval file. not sure why.

**Comment by erictraut** [comment]
I'd like to propose that we simply delete this command. It has clearly been broken for a long time, and no one has complained. I implemented it a long time ago before `strictParameterNoneValue` was on by default.
